comp/core/autodiscovery/scheduler/controller.go

```diff
@@ -43,9 +38,16 @@ type Controller struct {
 	stopChannel chan struct{}
 }
 
-// NewController inits a scheduler controller
+// NewController inits a scheduler controller without waiting, for unit test only
```

Comment:
is it for unit-test in the same package? If yes the function can be private and moved to a `_test.go` file.

If not, I would suggest to keep `NewController()` for the "official" function and rename this one: `NewAndStartController()`  to make it self explanatory.

Reply:
It's also for other packages 👍